saleor/order/notifications.py

```diff
@@ -0,0 +1,356 @@
+        "is_shipping_required": line.is_shipping_required,
+        "is_digital": line.is_digital,
+        "digital_url": digital_url,
+        "unit_discount_value": line.unit_discount_value,
+        "unit_discount_reason": line.unit_discount_reason,
+        "unit_discount_type": line.unit_discount_type,
+        "unit_discount_amount": line.unit_discount_amount,
+    }
+
+
+def get_lines_payload(order_lines: Iterable["OrderLine"]):
+    payload = []
+    for line in order_lines:
+        payload.append(get_order_line_payload(line))
+    return payload
+
+
+ADDRESS_MODEL_FIELDS = [
+    "first_name",
+    "last_name",
+    "company_name",
+    "street_address_1",
+    "street_address_2",
+    "city",
+    "city_area",
+    "postal_code",
+    "country",
+    "country_area",
+    "phone",
+]
+
+
+def get_address_payload(address):
+    if not address:
+        return None
+    address = model_to_dict(address, fields=ADDRESS_MODEL_FIELDS)
+    address["country"] = str(address["country"])
+    address["phone"] = str(address["phone"])
+    return address
+
+
+def get_discounts_payload(order):
+    order_discounts = order.discounts.all()
+    voucher_discount = None
+    all_discounts = []
+    discount_amount = 0
+    for order_discount in order_discounts:
+        dicount_obj = {
+            "type": order_discount.type,
+            "value_type": order_discount.value_type,
+            "value": order_discount.value,
+            "amount_value": order_discount.amount_value,
+            "name": order_discount.name,
+            "translated_name": order_discount.translated_name,
+            "reason": order_discount.reason,
+        }
+        all_discounts.append(dicount_obj)
+        if order_discount.type == OrderDiscountType.VOUCHER:
+            voucher_discount = dicount_obj
+        discount_amount += order_discount.amount_value
+
+    return {
+        "voucher_discount": voucher_discount,
+        "discounts": all_discounts,
+        "discount_amount": discount_amount,
+    }
+
+
+ORDER_MODEL_FIELDS = [
+    "id",
+    "token",
+    "display_gross_prices",
+    "currency",
+    "discount_amount",
+    "total_gross_amount",
+    "total_net_amount",
+    "undiscounted_total_gross_amount",
+    "undiscounted_total_net_amount",
+    "status",
+    "metadata",
+    "private_metadata",
+    "user_id",
+    "language_code",
+]
+
+
+def get_default_order_payload(order: "Order", redirect_url: str = ""):
+    order_details_url = ""
+    if redirect_url:
+        order_details_url = prepare_order_details_url(order, redirect_url)
+    subtotal = order.get_subtotal()
+    tax = order.total_gross_amount - order.total_net_amount
+
+    lines = order.lines.prefetch_related(
+        "variant__product__media",
+        "variant__media",
+        "variant__product__attributes__assignment__attribute",
+        "variant__product__attributes__values",
+    ).all()
+    order_payload = model_to_dict(order, fields=ORDER_MODEL_FIELDS)
```

Comment:
What do you thing to use here `OrderLineData`?

Reply:
Hhm, Could you explain the benefits of this?